test/test_ranker.py

```diff
@@ -1,123 +1,54 @@
-import math
-import numpy as np
 import os
-from tsg.config import DATA_DIR, RAW_DIR
+from tsg.config import DATA_DIR
 from tsg.ranker import get_dictionary_term_list, cosine_score_calc, calculate_query_term_weight
 
 def mock_dictionary_create() :
 	line1 = 'term c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3322237271982384,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3205763030575843,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3205763030575843\n'
 	line2 = 'to c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3205763030575843,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3322237271982384,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3205763030575843\n'
 	line3 = 'evaluate c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3205763030575843,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3205763030575843,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3322237271982384\n'
-	mock_filename = DATA_DIR + 'mock_dictionary.dat'
+
+	try:
+		os.remove(mock_filename)
+	except OSError:
+		pass
+	assert not os.path.exists(mock_filename)
 
 	with open(mock_filename,'w') as f:
 		f.write(line1)
 		f.write(line2)
 		f.write(line3)
 
+mock_filename = DATA_DIR + 'mock_dictionary.dat'
```

Comment:
Hmm this is not it.. sorry :)
http://nose.readthedocs.org/en/latest/writing_tests.html

Search in the page for 'with_setup'.
This is what you are looking for:

@with_setup(setup_func, teardown_func)
def func(arg):
    assert something_about(arg)